templates/common/_base/units/node-valid-hostname.service

```diff
@@ -0,0 +1,25 @@
+name: node-valid-hostname.service
+enabled: true
+contents: |
+  [Unit]
+  Description=Ensure hostname is not localhost
+  # Only run when the host has a localhost name.
+  ConditionHost=|localhost
+  ConditionHost=|localhost.localdomain
+  Before=network-online.target
+
+  [Service]
+  Type=oneshot
+  RemainAfterExit=yes
+  ExecStartPre=/bin/echo "Node has localhost hostname. Waiting for new hostname."
+  # Get the short hostname. This is more reliable than a regex.
+  ExecStartPre=/bin/bash -c 'while [ `hostname -s` == "localhost" ]; do sleep 1; done;'
+  ExecStart=/bin/sh -c "echo Node changed hostname to `hostname`"
+  # Wait up to 5min for the node to get a real hostname.
+  KillMode=process
+  TimeoutSec=300
+
+  [Install]
+  WantedBy=multi-user.target
```

Comment:
I don't think we need this - if `network-online.target` isn't requested then we shouldn't run.

Reply:
https://github.com/openshift/machine-config-operator/blob/master/templates/master/01-master-kubelet/_base/units/kubelet.service.yaml#L7 

The kublet is started _after_ network-online.target.

Reply:
Right, I'm talking about removing the
`WantedBy=multi-user.target`